src/Service/Doctrine/ResultProvider.php

```diff
@@ -317,7 +318,46 @@ private function calculateTotalCount(Pager $filter, int $resultCount)
     private function findCount(AnalysedQuery $analysedQuery): int
     {
         $countQueryBuilder = $analysedQuery->cloneQueryBuilder();
+        $groupByParts = $countQueryBuilder->getDQLPart('groupBy');
+
+        $this->validateGroupByParts($groupByParts);
+        if (count($groupByParts) === 1) {
+            $countQueryBuilder
+                ->resetDQLPart('groupBy')
+                ->select(sprintf('count(distinct %s)', $groupByParts[0]->getParts()[0]))
+            ;
+
+            $nullQueryBuilder = $analysedQuery->cloneQueryBuilder()
+                ->resetDQLPart('groupBy')
+                ->select($analysedQuery->getRootAlias())
+                ->setMaxResults(1)
+                ->andWhere($groupByParts[0]->getParts()[0] . ' is null')
+            ;
+
+            $nonNullCount = (int)$countQueryBuilder->getQuery()->getSingleScalarResult();
+            $nullExists = count($nullQueryBuilder->getQuery()->getScalarResult());
+
+            return  $nonNullCount + $nullExists;
+        }
```

Comment:
Can we move all the block to a separate private method? It seems like a thing that's concerting a level with more details – `findCount` will be clearer in that case (especially as group by is more an edge case, not default one)

Reply:
It can also be a bit clearer if our validation method (with some different naming) would return `$groupByParts[0]->getParts()[0]` for us. Currently we use it in several places, we need to look at the validation method to be sure that this will really always be available (and no part will have more than 1 item). Making this part of API (returning that string) would allow us to understand this (and maintain this behaviour) better.